dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/RMWAbstract.java

```diff
@@ -16,14 +15,14 @@ public abstract class RMWAbstract extends MemEvent implements RegWriter, RegRead
     protected final Register resultRegister;
     protected ExprInterface value;
 
-    RMWAbstract(IExpr address, Register register, ExprInterface value, String mo) {
+    protected RMWAbstract(IExpr address, Register register, ExprInterface value, String mo) {
```

Comment:
You shouldn't move this class to the `core` package. I would rather have it in `lang` (at top-level, not in any of its subpackages). The reason is that `RMWAbstract` is always getting compiled down to more primitive events and the `core` package should contain only those primitives.
Alternatively, you could keep the class in `lang.linux` and just create a new PTX-specific version (`PTXRMWAbstract`) in the ptx package.

We need to refactor the event package at some point to give a better place for common events like `RMWAbstract`, but for now I think the above two solutions are the best one's.

Reply:
`RMWAbstract` moved to `lang`